lib/matplotlib/axes/_axes.py

```diff
@@ -4338,6 +4340,7 @@ def _parse_scatter_color_args(c, edgecolors, kwargs, xsize,
         - kwargs['edgecolor']
         - edgecolors (is an explicit kw argument in scatter())
         - kwargs['color'] (==kwcolor)
+        - c (if not None)
```

Comment:
I think this is only true if `facecolors` is `"none"`. Otherwise I would expect the test case `(dict(c='b'), None),` from the edited test to instead be `(dict(c='b'), np.array([[0, 0, 1, 1]]))`, similar to the last test case added... I think, at least?